### PR TITLE
Added some modifications into the global functions, and fixed a spelling mistake

### PR DESCRIPTION
**Pull request by ZeeCoder**
- Fixed the success sentence, which was outputted when a new phar build was finished,
 - Added automatic environment variable registration to the `server` function ("host" and "name" variables),
 - Added a `has` function, so that the existence of a parameter can be checked.

I'm awaiting your opinions.

### CODE REVIEW

**Comment by ZeeCoder** [comment]
Forgot to mention, if this gets merged in, then the documentation should definitely mention the automatically generated env variables. (host/name)
Same with the `has` function, it must be listed in the "Functions" section of the docs.

File: src/functions.php [hunk 1]
```diff
@@ -51,6 +51,10 @@ function server($name, $domain = null, $port = 22)
     $deployer->servers->set($name, $server);
     $deployer->environments->set($name, $env);
 
+    // Setting environment variables about the server automatically
+    $env->set('name', $server);
+    $env->set('host', $host);
```
Comment (elfet) [inline src/functions.php]: I think it's better to put this lines inside `Builder` class, as there are `localServer` function needs this lines too.

Also can you add some test to check after function `server`/`localServer` call env vars a created?

**Comment by elfet** [comment]
And check new issues here: https://scrutinizer-ci.com/g/deployphp/deployer/inspections/fe07aa87-2431-4754-ab40-813f64962e04

File: src/functions.php [hunk 2]
```diff
@@ -51,6 +51,10 @@ function server($name, $domain = null, $port = 22)
     $deployer->servers->set($name, $server);
     $deployer->environments->set($name, $env);
 
+    // Setting environment variables about the server automatically
+    $env->set('name', $server);
```
Comment (elfet) [inline src/functions.php]: `$server` isn't string, it's `ServerInterface`. I think you mention: 
``` php
$env->set('name', $name);
```

**Comment by elfet** [comment]
BTW, its better to separate PR into small, independent ones. So it's simple to merge them.

**Comment by ZeeCoder** [comment]
I'll keep that in mind next time. :)
I'll check these issues as soon as I can.

**Comment by ZeeCoder** [comment]
I refactored some things and played around the "host" and "name" environment settings.
I discarded the latter idea because the first - naive - implementation would certainly cause issues in the future.

I'll open a separate issue about the env variables.

**Comment by oanhnn** [comment]
After 7 commits, this PR has:
- Spelling fix. Fixed the success sentence, which was outputted when a new phar build was finished,
- Added a `has` function, so that the existence of a parameter can be checked.
I think clear more if you re-base before merging code.

**Comment by elfet** [comment]
Do we really need `has` function?

**Comment by ZeeCoder** [comment]
I can break this PR into two separate one if you think, one for the spelling and an other for the has function.

I added that global function, since I wanted to check inside a task, whether a global configuration array was set or not. I wanted to implement proper error handling, which is as informative as it can be.
The current `get` implementation throws an exception if the given key does not exist in the collection, which is fine, but I wanted a different error message.
I can of course put the `get` call in a try-catch block, and throw my own exception, but then there's the chance of catching exceptions not related to the existence of the global variable I was trying to access.

The task I was trying to implement was a slack notification task, and with the `has` function I could do the following:

```php
// Checking for the existence of the slack task configuration.
// If it doesn't exist, I throw a custom exception with detailed information
// (Actually it could be even more detailed, by giving the expected array keys,
// but this is only an example.)
if (!has('slack')) {
    throw new \RuntimeException('Missing "slack" parameter for the "slack:push:success" task. Please provide it using the "set(\'slack\', [])" function.');
}

// The slack global variable now surely exist. If this get call throws
// something, that will not be related to the existence of the variable.
$config = get('slack');

// Validating the given configuration options goes here 
// ...
```

In short: I wanted to implement proper error handling inside a given task, and I think a `has` function can make that very simple.

**Comment by elfet** [comment]
Looks good. I'm merging.

**Comment by ZeeCoder** [comment]
Thanks. :)
On a related note: would you consider adding a simple "slack" task?
Because I could share what I have.

**Comment by elfet** [comment]
Take a look at: https://github.com/deployphp/recipes

**Comment by ZeeCoder** [comment]
Now that slack implementation is a perfect example why I wanted to access the "host" server configuration.
Does it work in v3?
It uses the `config()` function which I think does not exist anymore:
https://github.com/deployphp/recipes/blob/master/recipes/slack.php

**Comment by elfet** [comment]
Yeahh, need to create issue there.

**Comment by ZeeCoder** [comment]
On it. :)
